src/python/pants/backend/python/goals/setup_py.py

```diff
@@ -501,6 +502,10 @@ async def package_python_dist(
     working_directory = os.path.join(chroot_prefix, chroot.working_directory)
     prefixed_input = await Get(Digest, AddPrefix(input_digest, chroot_prefix))
     build_system = await Get(BuildSystem, BuildSystemRequest(prefixed_input, working_directory))
+    output_path = dist_tgt.get(PythonDistributionOutputPathField).value
+    assert (
+        output_path is not None
```

Comment:
@benjyw this assertion seemed awkward, but it was the only way I could see to convince mypy that `PythonDistributionOutputPathField` wouldn't be `None`.

Reply:
Totally fine to have this assert, for this reason. Mypy is a weirdo.